include/envoy/server/instance.h

```diff
@@ -164,6 +164,11 @@ class Instance {
    */
   virtual Stats::Store& stats() PURE;
 
+  /**
+   * @return the histogram statistics.
+   */
+  virtual std::list<Stats::HistogramStatistics>& histogramStats() PURE;
```

Comment:
Instead of hanging the histograms off of the server, can these come from `Stats::Store`? I think this would be better, even if the store potentially has to look into the sinks somehow (I haven't thought through this in depth).